BAU: Local running

Wynndow opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

This adds a new gradle sub-project to allow starting the lambdas up
locally in a Spark web app.

It also a Dockerfile and docker-compose file to start all the services
required up in containers, and for them to communicate with each other.

### Why did it change

To allow an option to debug lambdas, and to iterate faster when developing.

Wynndow committed:
BAU: Use input streams in state machine initializer

This updates the state machine initializer to fetch the journey map from
the resources as a stream, rather than a file.

When importing the process-journey-event lambda into a different project
(the local-running gradle project), loading as a file breaks. The file
path is a jar file path to the process-journey-event jar and for unknown
reasons the file can't be found withing the generated jar.

Switching to use streams works just fine.

Wynndow committed:
BAU: Add local running

This adds a new gradle sub-project to allow starting the lambdas up
locally in a Spark web app.

It also a Dockerfile and docker-compose file to start all the services
required up in containers, and for them to communicate with each other.

See the included README for more details.

lwebb-iig reviewed (commented):
Looks good - did you want to capture any of the additional thoughts / future work around some of the docker quirks mentioned yesterday? the README.md strikes me as a good place.
From memory these were:
- Docker log format
- Gradle docker image

Wynndow committed:
BAU: Add future improvements section to README

Wynndow committed:
BAU: Add logging configuration for local running

This adds a new log4j logging configuration to the local-running project
that means core-backs logs are all written in an easy to read way,
rather than JSON blobs.

The pattern can be tweaked if we want, the docs that explain the options
available are here: https://logging.apache.org/log4j/2.x/manual/layouts.html#pattern-layout

Wynndow committed:
BAU: Update proven identity details route

The endpoint on the internal API that the build proven user identity
details lambda sits behind has changes. This updates the local route to
match so it still works with the frontend.
